core/src/main/java/org/elasticsearch/cluster/metadata/MetaDataCreateIndexService.java

```diff
@@ -176,7 +178,12 @@ public void validateIndexName(String index, ClusterState state) {
         }
     }
 
-    public void createIndex(final CreateIndexClusterStateUpdateRequest request, final ActionListener<ClusterStateUpdateResponse> listener) {
+    public void createIndex(final CreateIndexClusterStateUpdateRequest request,
+                            final ActionListener<ClusterStateUpdateResponse> listener,
+                            final ThreadContext threadContext) {
```

Comment:
I think we should add ThreadPool as dependency to MetaDataCreateIndexService instead.